src/plugins/interactive_setup/server/kibana_config_writer.ts

```diff
@@ -54,31 +62,36 @@ export class KibanaConfigWriter {
   public async writeConfig(params: WriteConfigParameters) {
     const caPath = path.join(path.dirname(this.configPath), `ca_${Date.now()}.crt`);
 
-    this.logger.debug(`Writing CA certificate to ${caPath}.`);
-    try {
-      await fs.writeFile(caPath, params.ca);
-      this.logger.debug(`Successfully wrote CA certificate to ${caPath}.`);
-    } catch (err) {
-      this.logger.error(
-        `Failed to write CA certificate to ${caPath}: ${getDetailedErrorMessage(err)}.`
-      );
-      throw err;
+    if (params.caCert) {
+      this.logger.debug(`Writing CA certificate to ${caPath}.`);
+      try {
+        await fs.writeFile(caPath, params.caCert);
+        this.logger.debug(`Successfully wrote CA certificate to ${caPath}.`);
+      } catch (err) {
+        this.logger.error(
+          `Failed to write CA certificate to ${caPath}: ${getDetailedErrorMessage(err)}.`
+        );
+        throw err;
+      }
     }
 
     this.logger.debug(`Writing Elasticsearch configuration to ${this.configPath}.`);
     try {
+      const config: Record<string, any> = { 'elasticsearch.hosts': [params.host] };
```

Comment:
>  Config can be any valid JSON type. 

Well, we don't need to cover a general Config type, we want to just model type for our own use case, as strict as we can.

> since it adds little value here to be more specific.

The `any` usage in Kibana plugins is being tracked continuously with the goal to get rid of it completely (check out `Quality Report` in KIbana Contributors Newsletter). But even without that, it helps to avoid typos like accidentally passing object or anything like that.

Reply:
ill fix this in a follow up